src/codecs/webp/decoder.ts

```diff
@@ -0,0 +1,16 @@
+import { blobToArrayBuffer, imageDataToBitmap } from '../../lib/util';
+import DecoderWorker  from './Decoder.worker';
+
+export const name = 'WASM WebP Decoder';
+export const supportedExtensions = ['webp'];
+export const supportedMimeTypes = ['image/webp'];
+export async function decode(file: File): Promise<ImageBitmap> {
+  const decoder = await new DecoderWorker();
+  const imageData = await decoder.decode(await blobToArrayBuffer(file));
+  return imageDataToBitmap(imageData);
+}
+
+export async function isSupported(): Promise<boolean> {
+  // TODO(@surma): Should we do wasm detection here or something?
```

Comment:
Nah. WASM is part of our baseline. If we want to feature-detect this, it'll be right at the start of the app where we may choose to display a helpful message to unsupported browsers.